// css-tree/lib/syntax/create.js
import { tokenize } from 'css-tree/lib/tokenizer/index.js';
import { createParser } from 'css-tree/lib/parser/create.js';
import { createGenerator } from 'css-tree/lib/generator/create.js';
import { createConvertor } from 'css-tree/lib/convertor/create.js';
import { createWalker } from 'css-tree/lib/walker/create.js';
import { Lexer } from 'css-tree/lib/lexer/Lexer.js';
import mix from 'css-tree/lib/syntax/config/mix.js';

function createSyntax(config) {
    const parse = createParser(config);
    const walk = createWalker(config);
    const generate = createGenerator(config);
    const { fromPlainObject, toPlainObject } = createConvertor(walk);

    const syntax = {
        lexer: null,
        createLexer: config => new Lexer(config, syntax, syntax.lexer.structure),

        tokenize,
        parse,
        generate,

        walk,
        find: walk.find,
        findLast: walk.findLast,
        findAll: walk.findAll,

        fromPlainObject,
        toPlainObject,

        fork(extension) {
            const base = mix({}, config); // copy of config

            return createSyntax(
                typeof extension === 'function'
                    ? extension(base, Object.assign)
                    : mix(base, extension)
            );
        }
    };

    syntax.lexer = new Lexer({
        generic: true,
        units: config.units,
        types: config.types,
        atrules: config.atrules,
        properties: config.properties,
        node: config.node
    }, syntax);

    return syntax;
};

export default config => createSyntax(mix({}, config));
