// css-tree/lib/syntax/create.js
var List = require('css-tree/lib/common/List');
var SyntaxError = require('css-tree/lib/common/SyntaxError');
var TokenStream = require('css-tree/lib/common/TokenStream');
var Lexer = require('css-tree/lib/lexer/Lexer');
var definitionSyntax = require('css-tree/lib/definition-syntax');
var tokenize = require('css-tree/lib/tokenizer');
var createParser = require('css-tree/lib/parser/create');
var createGenerator = require('css-tree/lib/generator/create');
var createConvertor = require('css-tree/lib/convertor/create');
var createWalker = require('css-tree/lib/walker/create');
var clone = require('css-tree/lib/utils/clone');
var names = require('css-tree/lib/utils/names');
var mix = require('css-tree/lib/syntax/config/mix');

function createSyntax(config) {
    var parse = createParser(config);
    var walk = createWalker(config);
    var generate = createGenerator(config);
    var convert = createConvertor(walk);

    var syntax = {
        List: List,
        SyntaxError: SyntaxError,
        TokenStream: TokenStream,
        Lexer: Lexer,

        vendorPrefix: names.vendorPrefix,
        keyword: names.keyword,
        property: names.property,
        isCustomProperty: names.isCustomProperty,

        definitionSyntax: definitionSyntax,
        lexer: null,
        createLexer: function(config) {
            return new Lexer(config, syntax, syntax.lexer.structure);
        },

        tokenize: tokenize,
        parse: parse,
        walk: walk,
        generate: generate,

        find: walk.find,
        findLast: walk.findLast,
        findAll: walk.findAll,

        clone: clone,
        fromPlainObject: convert.fromPlainObject,
        toPlainObject: convert.toPlainObject,

        createSyntax: function(config) {
            return createSyntax(mix({}, config));
        },
        fork: function(extension) {
            var base = mix({}, config); // copy of config
            return createSyntax(
                typeof extension === 'function'
                    ? extension(base, Object.assign)
                    : mix(base, extension)
            );
        }
    };

    syntax.lexer = new Lexer({
        generic: true,
        types: config.types,
        atrules: config.atrules,
        properties: config.properties,
        node: config.node
    }, syntax);

    return syntax;
};

exports.create = function(config) {
    return createSyntax(mix({}, config));
};
