// css-tree/cjs/index.cjs
'use strict';

const index$1 = require('css-tree/cjs/syntax/index.cjs');
const version = require('css-tree/cjs/version.cjs');
const create = require('css-tree/cjs/syntax/create.cjs');
const List = require('css-tree/cjs/utils/List.cjs');
const Lexer = require('css-tree/cjs/lexer/Lexer.cjs');
const index = require('css-tree/cjs/definition-syntax/index.cjs');
const clone = require('css-tree/cjs/utils/clone.cjs');
const names$1 = require('css-tree/cjs/utils/names.cjs');
const ident = require('css-tree/cjs/utils/ident.cjs');
const string = require('css-tree/cjs/utils/string.cjs');
const url = require('css-tree/cjs/utils/url.cjs');
const types = require('css-tree/cjs/tokenizer/types.cjs');
const names = require('css-tree/cjs/tokenizer/names.cjs');
const TokenStream = require('css-tree/cjs/tokenizer/TokenStream.cjs');

const {
    tokenize,
    parse,
    generate,
    lexer,
    createLexer,

    walk,
    find,
    findLast,
    findAll,

    toPlainObject,
    fromPlainObject,

    fork
} = index$1;

exports.version = version.version;
exports.createSyntax = create;
exports.List = List.List;
exports.Lexer = Lexer.Lexer;
exports.definitionSyntax = index;
exports.clone = clone.clone;
exports.isCustomProperty = names$1.isCustomProperty;
exports.keyword = names$1.keyword;
exports.property = names$1.property;
exports.vendorPrefix = names$1.vendorPrefix;
exports.ident = ident;
exports.string = string;
exports.url = url;
exports.tokenTypes = types;
exports.tokenNames = names;
exports.TokenStream = TokenStream.TokenStream;
exports.createLexer = createLexer;
exports.find = find;
exports.findAll = findAll;
exports.findLast = findLast;
exports.fork = fork;
exports.fromPlainObject = fromPlainObject;
exports.generate = generate;
exports.lexer = lexer;
exports.parse = parse;
exports.toPlainObject = toPlainObject;
exports.tokenize = tokenize;
exports.walk = walk;
