// @parcel/utils/src/index.js
// @flow strict-local
export type * from './config';
export type * from './Deferred';
export type * from './generateBuildMetrics';
export type * from './http-server';
export type * from './path';
export type * from './prettyDiagnostic';
export type * from './schema';

export {default as countLines} from '@parcel/utils/src/countLines';
export {default as generateBuildMetrics} from '@parcel/utils/src/generateBuildMetrics';
export {default as generateCertificate} from '@parcel/utils/src/generateCertificate';
export {default as getCertificate} from '@parcel/utils/src/getCertificate';
export {default as getModuleParts} from '@parcel/utils/src/getModuleParts';
export {default as getRootDir} from '@parcel/utils/src/getRootDir';
export {default as isDirectoryInside} from '@parcel/utils/src/isDirectoryInside';
export {default as isURL} from '@parcel/utils/src/is-url';
export {default as objectHash} from '@parcel/utils/src/objectHash';
export {default as prettifyTime} from '@parcel/utils/src/prettifyTime';
export {default as prettyDiagnostic} from '@parcel/utils/src/prettyDiagnostic';
export {default as PromiseQueue} from '@parcel/utils/src/PromiseQueue';
export {default as validateSchema} from '@parcel/utils/src/schema';
export {default as TapStream} from '@parcel/utils/src/TapStream';
export {default as urlJoin} from '@parcel/utils/src/urlJoin';
export {default as relativeUrl} from '@parcel/utils/src/relativeUrl';
export {default as createDependencyLocation} from '@parcel/utils/src/dependency-location';
export {default as debounce} from '@parcel/utils/src/debounce';
export {default as throttle} from '@parcel/utils/src/throttle';
export {default as openInBrowser} from '@parcel/utils/src/openInBrowser';

// Explicit re-exports instead of export * for lazy require performance
export {findAlternativeNodeModules, findAlternativeFiles} from '@parcel/utils/src/alternatives';
export {blobToBuffer, blobToString} from '@parcel/utils/src/blob';
export {
  unique,
  objectSortedEntries,
  objectSortedEntriesDeep,
  setDifference,
  setEqual,
  setIntersect,
  setUnion,
} from '@parcel/utils/src/collection';
export {
  resolveConfig,
  resolveConfigSync,
  loadConfig,
  readConfig,
} from '@parcel/utils/src/config';
export {DefaultMap, DefaultWeakMap} from '@parcel/utils/src/DefaultMap';
export {makeDeferredWithPromise} from '@parcel/utils/src/Deferred';
export {getProgressMessage} from '@parcel/utils/src/progress-message.js';
export {
  isGlob,
  isGlobMatch,
  globMatch,
  globSync,
  glob,
  globToRegex,
} from '@parcel/utils/src/glob';
export {hashStream, hashObject, hashFile} from '@parcel/utils/src/hash';
export {SharedBuffer} from '@parcel/utils/src/shared-buffer';
export {fuzzySearch} from '@parcel/utils/src/schema';
export {createHTTPServer} from '@parcel/utils/src/http-server';
export {normalizePath, normalizeSeparators, relativePath} from '@parcel/utils/src/path';
export {
  replaceURLReferences,
  replaceInlineReferences,
} from '@parcel/utils/src/replaceBundleReferences';
export {
  measureStreamLength,
  readableFromStringOrBuffer,
  bufferStream,
  blobToStream,
  streamFromPromise,
  fallbackStream,
} from '@parcel/utils/src/stream';
export {relativeBundlePath} from '@parcel/utils/src/relativeBundlePath';
export {ansiHtml} from '@parcel/utils/src/ansi-html';
export {escapeHTML} from '@parcel/utils/src/escape-html';
export {
  SOURCEMAP_RE,
  SOURCEMAP_EXTENSIONS,
  matchSourceMappingURL,
  loadSourceMapUrl,
  loadSourceMap,
  remapSourceLocation,
} from '@parcel/utils/src/sourcemap';
export {default as stripAnsi} from 'strip-ansi';
