// csso/cjs/replace/index.cjs
'use strict';

const cssTree = require('css-tree');
const Atrule = require('csso/cjs/replace/Atrule.cjs');
const AttributeSelector = require('csso/cjs/replace/AttributeSelector.cjs');
const Value = require('csso/cjs/replace/Value.cjs');
const Dimension = require('csso/cjs/replace/Dimension.cjs');
const Percentage = require('csso/cjs/replace/Percentage.cjs');
const _Number = require('csso/cjs/replace/Number.cjs');
const Url = require('csso/cjs/replace/Url.cjs');
const color = require('csso/cjs/replace/color.cjs');

const handlers = {
    Atrule,
    AttributeSelector,
    Value,
    Dimension,
    Percentage,
    Number: _Number.Number,
    Url,
    Hash: color.compressHex,
    Identifier: color.compressIdent,
    Function: color.compressFunction
};

function replace(ast) {
    cssTree.walk(ast, {
        leave(node, item, list) {
            if (handlers.hasOwnProperty(node.type)) {
                handlers[node.type].call(this, node, item, list);
            }
        }
    });
}

module.exports = replace;
